### DISCUSSION TITLE
Communications about JdeRobot projects and activities (including GSoC)

### DISCUSSION THREAD

**Discussion by pariaspe** [JdeRobot/jderobot.github.io#233] [Announcements]
Hi folks,

as previously discussed, we are creating a more generic discussion page not strictly linked to RoboticsAcademy project. This channel is mainly **a tool for developer communications, conversations, discussions around  [JdeRobot](https://jderobot.github.io/) projects and activities**, such as GSoC. 

Besides, this channel is **a tool for interacting with other members of the community**. It is open to new users and developers of JdeRobot's. Don't be shy, just use it and share your questions and proposals here.

Welcome to the JdeRobot Discussions!

**Comment by Samkhan705279** [comment]
hello, myself umar ali khan btech 2 nd year electronics engineering student. this time I am also going  to participate in gsoc 2024.

**Reply by jmplaza** [reply]
Nice to meet you @Samkhan705279 . 

A good way to start contributing is to get familiar with our software products. In particular, RoboticsAcademy is the most active one and a priority project. The best way to start contributing is to **USE RoboticsAcademy** (solve and provide feedback) about the exercises already available in [current release](https://jderobot.github.io/RoboticsAcademy/exercises/) (FollowLine, ObstacleAvoidance, VacuumCleaner, LocalizedVacuumCleaner...). They use the browser as the only GUI (for source code editing and for exercise monitoring), and a docker image for running the Gazebo simulator (it is named RADI= Robotics Applications Docker Image). The docker image already includes a Django webserver for providing the exercise webpages. Please use the [RoboticsAcademy Discussions](https://github.com/JdeRobot/RoboticsAcademy/discussions) for questions (you will receive support there) and report any detected bug creating an issue for each one in the [RoboticsAcademy repository](https://github.com/JdeRobot/RoboticsAcademy). Feel free to share videos in the forum or mentioning @JdeRobot in twitter of your solutions to those exercises in operation. It also includes some exercises on DeepLearning: digit classifier and human detection.

After using RoboticsAcademy, a second step in contributing is studying its source code and **sending PullRequests which solve any of the open issues**. 

Other active JdeRobots projects are [VisualCircuit](https://github.com/JdeRobot/VisualCircuit) ([demo video](https://www.youtube.com/watch?v=WMRDqv3TYB0)) and [FPGA-Robotics](https://github.com/JdeRobot/FPGA-robotics) ([demo video](https://www.youtube.com/watch?v=rbdQ36ZJ7Lo)). We have one Slack channel per project and several WorkingGroups for solid contributors.

Cheers!

**Reply by Samkhan705279** [reply]
@jmplaza 
I will definitely look into this, and reach out if I got struck at any place.

**Reply by izzaaaatanishq** [reply]
Hello @jmplaza,

I have a question about resolving open issues. Suppose I've worked on an issue and made changes to the code. I'm curious about how changes made to the code for resolving open issues are reflected in the RADI or Docker image, given that the changes are made in the GitHub repository.

**Reply by jmplaza** [reply]
The received PullRequests (PR) are reviewed by a committer (=experienced developer). In negative case he/she will give feedback to the contributor. In positive case the PR will be merged at the GitHub repository master branch. Periodically we produce new RADI releases, which are build from that branch.

Currently we are pretty busy in the upcoming RoboticsAcademy release (5.2), which includes a large refactoring, but eventually we will publish it, with the corresponding new RADI releases too.

Meanwhile you can work with your locally generated mini-RADI, which may include your changes.

**Comment by zeeshan8302** [comment]
Hello, my name is Zeeshan and I'm passionate about Python . I'm currently appearing for GSoC 2024 and I'm always looking to connect with others in this Organization . It's a pleasure to meet you all!

And from where I should start.

**Reply by jmplaza** [reply]
Hi @zeeshan8302 ,

JdeRobot will apply for GSoC-2024, but it has not been selected yet. In fact, the org selection process from GSoC has not started yet.  In case of being selected again for GSoC, the candidate selection process will be similar to the one we followed in 2022 and [2023](https://jderobot.github.io/activities/gsoc/2023#application-instructions-for-gsoc-2023). In addition, in ChatGPT times, this year we will take a closer look at the previous contributions of the candidates to any JdeRobot project, sent Pull Requests, activity in the repos, etc.

Regarding the starting point, take a look at the previous message in this thread.

**Comment by aniketsona23** [comment]
Hello, I am Aniket Sonawane and fluent in python, C++ and slight in MERN stack, I am new to Open source and have worked somewhat with ROSNoetic, I am really interested in learning more and contributing to this Organization.

**Reply by jmplaza** [reply]
Hi @aniketsona23 , take a look at this [previous response](https://github.com/orgs/JdeRobot/discussions/233#discussioncomment-7983227) about how to start contributing to JdeRobot org.

**Comment by Samkhan705279** [comment]
---------- Forwarded message ---------
From: 'sttaylor' via Google Summer of Code Announce <
***@***.***>
Date: Fri, 5 Jan, 2024, 4:56 am
Subject: GSoC Announcement: GSoC Organization Application Info Session
January 11th 1700 UTC
To: Google Summer of Code Announce <
***@***.***>


Happy New Year!

We are very excited to get started on this 20th year of Google Summer of
Code!  With Organization applications opening in a few weeks, January 22 -
February 6, we wanted to host an information session for organizations
looking to apply to GSoC as well as for new org admins for veteran orgs so
they understand the application process.

Date: Thursday, January 11

Time: 17:00 - 17:45 UTC

Meet Link: meet.google.com/ogc-drzu-prz

GSoC Program Lead, Stephanie Taylor will go through the org application
process and discuss the tips to a solid organization application as well as
some of the other things to consider when applying as a GSoC organization.

There will be plenty of time for Q&A as well for the last half of the
session.

Some quick tips for everyone to consider when submitting an Organization
application for Google Summer of Code:

   1.

   Starting in 2024 we will have small (~90 hour projects), medium (~175 hr
   projects) and large (~350 hour) projects available to GSoC contributors.
   Orgs should have medium and large projects available in their Project Ideas
   lists. Small project ideas are not required for orgs, but if the smaller
   size project works for your org they should  be included in your
   Organization’s Ideas List.
   2.

   Reach out to your community members now to ask if they would like to be
   mentors for the program.

Having a thorough and well thought out list of Project Ideas
<https://google.github.io/gsocguides/mentor/defining-a-project-ideas-list>
is the most important part of your application.

You can check out the GSoC Org Application Short Answer Questions
<https://docs.google.com/document/d/1OL62tQtj72NB94gsEiu8DsqQGmCJgnhVKiKWXruGd0o/edit?usp=sharing>
if you’d like to start preparing your responses in advance so you can copy
over your answers once the org applications open January 22 at g.co/gsoc.

Please encourage other open source orgs to apply -- if you know of other
open source projects that may be interested in applying to GSoC as a first
time org please remind them to check out the available resources below and
have them put your org (or you) down as a reference when they apply.


We are also looking to add more AI/ML open source organizations in 2024 as
GSoC mentoring orgs, so please spread the word to your colleagues.

Open source projects can apply <https://summerofcode.withgoogle.com/> to be
mentoring organizations from January 22 - February 6 at 1800 UTC.

Resources:

Mentor Guide <https://google.github.io/gsocguides/mentor/>

Timeline <https://developers.google.com/open-source/gsoc/timeline>

FAQs <https://developers.google.com/open-source/gsoc/faq>

Roles and Responsibilities
<https://developers.google.com/open-source/gsoc/help/responsibilities>

Marketing Materials
<https://developers.google.com/open-source/gsoc/resources/marketing> (slide
deck, flyers)

Videos <https://developers.google.com/open-source/gsoc/videos>


Best,

Stephanie Taylor

GSoC Program Lead

-- 
You received this message because you are subscribed to the Google Groups
"Google Summer of Code Announce" group.
To unsubscribe from this group and stop receiving emails from it, send an
email to ***@***.***
To view this discussion on the web visit
https://groups.google.com/d/msgid/google-summer-of-code-announce/e57b1555-4109-4296-a181-faa0a294276fn%40googlegroups.com
<https://groups.google.com/d/msgid/google-summer-of-code-announce/e57b1555-4109-4296-a181-faa0a294276fn%40googlegroups.com?utm_medium=email&utm_source=footer>
.

**Comment by AnushkaGharage** [comment]
Greetings everyone!

I'm Anushka Gharage!
I am a Third year Computer Science (Data Science) student at KITCOEK. 
I have worked on projects like 
    - Map Navigation using Google Maps API(HTML, CSS, JavaScript, Google Maps JavaScript API)
    - Fashion Recommendation System (Python, TensorFlow, Resnet50, CNN, Deep Learning)
However, what truly gets my gears turning is contributing to open-source projects and being part of the collaborative force that propels innovation.
I got interested in JdeRobot projects while browsing for GSOC 2024.
I'm here not just to soak up knowledge but to actively engage and contribute. If you have any exciting open-source projects, or if you're looking for an extra pair of hands, count me in! Let's collaborate, exchange ideas, and build something awesome together.

**Reply by jmplaza** [reply]
Hi @AnushkaGharage ,

https://github.com/orgs/JdeRobot/discussions/233#discussioncomment-7983733
https://github.com/orgs/JdeRobot/discussions/233#discussioncomment-7983227

**Comment by AnushkaGharage** [comment]
After going through description of all projects, I am interested in 'Project 4 Robotics-Academy: improve Deep Learning based Human Detection exercise'.
Under application instructions for GSOC for Project-4, I need to submit -
  RoboticsAcademy Challenge, Python challenge, ROS2 challenge (mandatory)
  C++ challenge, React challenge (optative)
  
By what date should I submit these challenges?
Is it okay if we submit the Youtube link here or is it mandatory to tweet the link?

**Reply by pawanw17** [reply]
Hi @AnushkaGharage, The Projects and challenges listed were for GSoC 2023 and not 2024, I'd suggest you wait for the official release of GSoC 2024 projects and challenges (If we are selected this year). Until then you could get involved in the community by using our Software (RoboticsAcademy, VisualCircuit, BehaviourMetrics etc) and reporting/fixing bugs or giving feature suggestions.

**Reply by Samkhan705279** [reply]
@pawanw17 i am pretty sure this time also jde robotics will get this opportunity.😊

**Reply by AnushkaGharage** [reply]
@pawanw17 Got it! I will get on with the software.

**Reply by pawanw17** [reply]
You can alsocontribute by fixing any deprecated documentation or typos or making our websites better.

**Comment by obamwonyi** [comment]
Hi , I am Destiny . I am a computer engineering undergrad fluent in C, Python  and PHP. I am also very familiar with HTML, CSS and Javascript along with the Laravel and Vuejs framework . I am new to opensource an would love to contribute in any way I can . You can learn more about from the links below 
linkedin : https://www.linkedin.com/in/destiny-obamwonyi-3a8a52218/
github : https://github.com/obamwonyi

**Comment by BkPankaj** [comment]
Hello everyone! I'm new to this discussion and excited to be a part of it.  
I'm Pankaj Borade, a final year Electronics & Communication Engineering student at the National Institute of Technology Surat. My journey so far has been a dynamic blend of programming, robotics, and web technology.

Skills:

Programming Languages: C++, Java, Python
Web Technologies: MERN Stack, JavaScript, HTML, CSS, WebRTC, Sockets, TypeScript
Robotics: ROS, ESP32, STM32, Arduino, KiCAD
Computer Science Concepts: OOP, DBMS, Git

Project Highlights:

Developed the Shiva 3.4 Humanoid Robot
Created the "Lift to Lift" Game using UNITY3D, C#, and Blender (800+ downloads on the Play Store)
Developed the StudentConnect Hub and One Place, web apps for student-college interaction and live messaging
Explored Deep Learning for handwritten number image identification
Represented NIT Surat in the DD Robocon National Competition

Achievements:

Three-time National Karate Champion and Second Dan Black Belt.

Internship Experience:

Integrated Standby mode and MODBUS in STM32
Developed a web interface using ReactJs, working on robot interaction using Node.js and Python middleware connected to ROS2 through Web Sockets and API.
Developing live camera streaming from Robot to UI through middleware using WebRTC.

Current Focus:

Completed setup for RoboticsAcademy and VisualCircuit
Finding issues and adding new features

Looking forward to learning from all of you and contributing to projects and GSoC 2024.

**Comment by BkPankaj** [comment]
Hello,
I have a question about VisualCircuit. Should I start the discussion here, or at VisualCircuit discussion? As, Currently, there seem to be no ongoing discussions or comments in the VisualCircuit discussion section.

**Reply by jmplaza** [reply]
Hi @BkPankaj ,
feel free to start the VisualCircuit discussion :-)

**Reply by marcospaulo429** [reply]
Hi @jmplaza, I'm Marcos and currently I'm starting to work in BehaviorMetrics. BehaviorMetrics doesn't have a discussion page too. If you could create a discussion tab there I would appreciate it

**Reply by jmplaza** [reply]
Sure @marcospaulo429 , done! It is available at [https://github.com/JdeRobot/BehaviorMetrics/discussions](https://github.com/JdeRobot/BehaviorMetrics/discussions)

**Reply by sergiopaniego** [reply]
Let's discuss the ideas in the new channel @marcospaulo429 :smile:

**Comment by Avisheet** [comment]
Hi,
After going through the projects of JdeRobot in GSOC 2024 I am interested in Project 1 ,2 and 10 . But I'm new and don't have much idea about how to get started in GSOC . I have a good knolwdge of ROS , ROS2 , Gazebo , Rviz, Python and moveit . Can you tell me how to proceed further for GSOC 2024?

**Reply by jmplaza** [reply]
Hi @Avisheet , just read the documentation :-) For instance [previous messages in this Discussions](https://github.com/orgs/JdeRobot/discussions/233#discussioncomment-7983227) and [our GSoC-2024 webpage](https://jderobot.github.io/activities/gsoc/2024)

**Reply by Avisheet** [reply]
@jmplaza I’m Avisheet Srivastava, an undergrad from NIT Hamirpur, India and studying in my sophomore year of Computer Science and Engineering (Integrated M.tech).I’m following the content and I’m interested in 2 ideas :

    Robotics-Academy: CI & Testing
    Robotics Academy: improvement of Gazebo scenarios and robot models
    Robotics Academy: improvement of autonomous driving exercises

I’m C.S.E guy and also do competitive Programming in Python and C++. In Past, I have done many work in ROS/ROS2,moveit2,Gazebo and solid works. I also took part in eYantra-23 and completed the simulation tasks.

Please tell me which one should I prefer according to the need of Org.

Any further suggestion and guidance would be really appreciable.

Thanks

**Reply by jmplaza** [reply]
@Avisheet  All the proposed projects are interesting for our Org. Just follow the directions in the links I've just shared in my response:  use RoboticsAcademy, send pull requests, contact the potential mentors, submit your programming tests.... This year, in chatGPT times, we will take contributions at our repos much into account of our candidate selection process for GSoC.

**Comment by BkPankaj** [comment]
Greeting everyone

I'm interested in some projects and have some doubts.

In Project 5: VisualCircuit: block library, Contributor have to build various designs using blocks and push them to a new GitHub repository. Eventually, this repository will become a block library. That block library will be accessible through a web interface, allowing students or users to access designs? Similar to [this.](https://www.multisim.com/discover/) Is my understanding correct? @jmplaza @toshan-luktuke 

Project 6: Robotics Academy: Exercise Builder seems quite intriguing as it will create a dynamic environment for students to collaboratively develop exercises and learn together. This will be a completely new project, right? It will have a tab in Robotics Academy where users can click to create exercises and deploy them for others, utilizing an architecture similar to Scratch programming or VisualCircuit blocks, but for exercise creation. Am I understanding this correctly? @Apoorvgarg-creator , David Valladares 

Regarding the programming challenges, mentioned GSOC 2022, are these challenges set to be updated, or do we need to submit that only?

Thanks & Regards
Pankaj Borade

**Reply by Apoorvgarg-creator** [reply]
Hello @BkPankaj, for Exercise Builder, We are aiming to create an platform via which we can streamline the process of Exercise creation. Currently to make an exercise, An exercise developer needs to have an in-depth knowledge of our Robotics Academy Architecture and then only they can start making an exercise. If you notice the source code for many exercises, they have a common underlying principle on which they are constructed. Using that and Incorporating these principles, We aim to provide this platform creating exercises via beautiful UI and No/low code solution.

I hope this helps. Please feel free to ask any questions.

**Reply by toshan-luktuke** [reply]
Hi @BkPankaj, currently we're not planning to build a separate website for the block library. However we'll be working on a workflow to add it to our [visualcircuit-resources](https://github.com/JdeRobot/VisualCircuit-resources) repository and documentation for budding developers. In addition we'll be focusing on automated testing and deployment of these features.

**Reply by BkPankaj** [reply]
@Apoorvgarg-creator  Thank you for the detailed answer, I truly appreciate it. Your explanation has addressed all of my doubts.

**Reply by BkPankaj** [reply]
@toshan-luktuke  Thank you for the reply. I have a question regarding the "workflow for adding it to our visualcircuit-resources repository." Does this mean that the contributor is expected to create multiple designs for block library during the coding period?

**Reply by toshan-luktuke** [reply]
No, we only need one design for the block library. By workflow I'm referring to the process of adding blocks to the block-library. It will need monitoring from our end to ensure the blocks are working. 
Having a clear and systematic workflow for this process will help both the contributors and maintainers in working more efficiently.
Hope this clears your doubt, if you have any more questions please feel free to ask

**Reply by BkPankaj** [reply]
@toshan-luktuke , Thanks for your explanation. It cleared up all my questions.

**Comment by TahsinOP** [comment]
Hello everyone , 
i am Tahsin Khan a 3rd year undergraduate at National Institute of Technology Karnataka , Surathkal ( NITK ) . I am a robotics enthusiast with a substantial experience in ROS2 , SLAM , autonomous navigation ( nav2 ) , arm manipulation and computer vision . I have done previous internships @ IIT D IAFSM involving simulating a UR5 arm for pick and place operations , also developed a end-to-end ROS2 based warehouse automation algorithm  involving ( package sorting and detection using CV , arm manipulation , SLAM based navigation ) for a e-YRC 2023 which is a national robotics competition hosted by IIT Bombay , our team completed the competition at an impressive #16th position

 I am very much interested to contribute to JdeRobot in GSoC - 2024 . I am specifically interested in **Project # 7**  Robotics Academy: improvement of Gazebo scenarios and robot models and **Project # 8** Robotics Academy: improvement of autonomous driving exercises . I have started setting up the Robot Academy interface and would love to contribute to the organization .
 
 Feel free to reach me on : 
 Email : tahsinkhan.211me352@nitk.edu.in 
Linkedin : [Tahsin Khan](https://www.linkedin.com/in/tahsin-khan-3aa4a4240/)

**Comment by Wei-cyber** [comment]
Hello everyone,

I am a computer science student and find all project ideas fascinating. I am familiar with machine learning, ROS2, Python, some experience in web development. I am looking forward to contributing to the organization and learning from each other.

**Comment by AryanNanda17** [comment]
Hello JdeRobot Community, 
I have been active in Jde Robot since January and want to contribute to it through GSOC. 
- I have done the setup of BehaviorMetrics and RoboticsAcademy. 
- I have completed all the Tasks mentioned in the ideas list page of 2024. 
- @dpascualhe and @pankhurivanjani, I am interested in the project Robotics-Academy: new exercise using Deep Learning for Visual Control, I have mailed you but I haven't received any response. So, please check out my email(nandaaryan823@gmail.com). 
I have sent you a GitHub invite to my repository containing all the challenges. So, please review the tasks and guide me on how to proceed further.

**Reply by jmplaza** [reply]
Hi @AryanNanda17, Nice to see your progress with BehaviorMetrics and with RoboticsAcademy. Watch out! we are updating (hopefully this week) the programming tests for GSoC-2024. Sorry for the delay.

**Reply by AryanNanda17** [reply]
Oh Okay @jmplaza. 
Thanks for the update!

**Comment by Wei-cyber** [comment]
Hello JdeRobot Community,
I have successfully downloaded the latest robotics academy docker image. I am doing local navigation with VFF now, but the simulation is not working (it becomes blank) on my end after loading. it was fine when I run it for the first time. I removed and then pulled the image again. the issue persists. I wonder if anyone can help? thanks
![Screenshot](https://github.com/JdeRobot/jderobot.github.io/assets/83705165/b12b057f-d649-4aed-b645-b4e2f16a25bc)

**Reply by jmplaza** [reply]
Please @Wei-cyber, use RADI 3.4.30 and move this comment better to [RoboticsAcademy Discussions](https://github.com/JdeRobot/RoboticsAcademy/discussions). Do you have a GPU on your machine? have you launched the RADI with nvidia GPU support.

**Reply by Wei-cyber** [reply]
Sorry about that. I have copied it to the robot academy discussion. Or is it better to reference the comment in the new issue?

**Comment by MihirGore23** [comment]
Hello community !

I am Mihir Gore, an undergrad from VJTI, Mumbai and I am really keen to contribute through GSoC. I am been active on Behavior Metrics and Robotics Academy since December, and I have worked on a PR in Behavior Metrics.

I have completed all the tasks mentioned on ideas list page, filled and submitted the google form.

I have setup both Behavior Metrics and Robotics Academy , and I am interested to work on 
Project 3 : Improving machine learning exersices ,
Project 4 : new exersice using ML for visual control.

I have mailed @pankhurivanjani and @dpascualhe regarding the same. 

I am facing certain issues with running dl_classifier and human_detection ML exersices locally, with them being not present in latest RADI. Looking forward to actively contributing, and hoping for guidance to resolve the issue.
